cloudify_agent/resources/script/windows.ps1.template

```diff
@@ -4,28 +4,35 @@ $ErrorActionPreference = "Stop"
 
 function Download($Url, $OutputPath)
 {
-    New-Item -ItemType directory -Path (Split-Path $OutputPath)
+    # Disable server side certificate validation (client side validations still exist)
+    [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
 
-    $WebClient = New-Object System.Net.WebClient
-    $WebClient.Headers.add('{{ auth_token_header }}', '{{ auth_token_value }}')
-    $WebClient.DownloadFile("$Url", "$OutputPath")
-}
+    # Make sure the output directory exists
+    New-Item -ItemType directory -Force -Path (Split-Path $OutputPath)
 
-function PatchCloudifyAgent($Branch)
-{
-    $DownloadUrl = "https://github.com/cloudify-cosmo/cloudify-agent/archive/$Branch.zip"
-    $DownloadOutputPath = "{{ conf.basedir }}\cloudify-agent.zip"
-    $ExtractedTo = "{{ conf.basedir }}\cloudify-agent-git"
-    Download $DownloadUrl $DownloadOutputPath
-    New-Item -ItemType directory -Path $ExtractedTo
-    Add-Type -Assembly "System.IO.Compression.FileSystem"
-    [IO.Compression.ZipFile]::ExtractToDirectory($DownloadOutputPath, $ExtractedTo)
-    Copy-Item "$ExtractedTo\cloudify-agent-$Branch\cloudify_agent" "{{ conf.agent_dir }}\Lib\site-packages" -Recurse -Force
+    $WebRequest = [System.Net.WebRequest]::Create($Url)
+
+    # Create certificate object from file and add it to the HTTP request
+    $Cert = [System.Security.Cryptography.X509Certificates.X509Certificate2]::CreateFromCertFile('{{ ssl_cert_path }}')
+    $WebRequest.ClientCertificates.Add($Cert)
+
+    $WebRequest.Headers.Add('{{ auth_token_header }}', '{{ auth_token_value }}')
+
+    # Create a stream from the HTTP response and create the output file
+    $ResponseStream = $WebRequest.GetResponse().GetResponseStream()
+    $OutputFile = [System.IO.File]::Create($OutputPath)
+    $Buffer = New-Object Byte[] 1024
+
+    # Write the stream to the output file in 1KB chunks
+    While (($BytesRead = $ResponseStream.Read($Buffer, 0, $Buffer.Length)) -gt 0) {
+        $OutputFile.Write($Buffer, 0, $BytesRead)
+    }
+    $OutputFile.Close()
```

Comment:
It probably doesn't matter much, but you can use the Try..Finally pattern here and call $OutputFile.Close() in the finally block.